### PR TITLE
Add the analogy of pull requests and editing a shared document

### LINKED ISSUE #463: Improve introduction of pull requests
### How could the content be improved?

We introduce the concept of branches just before the pull request demo+exercise, and I still find it difficult to do this elegantly and efficiently, e.g. judging the right level of detail to provide that will make sense without getting stuck in the weeds of a topic that can be conceptually rather heavy. I would like to expand/polish up the introductory paragraph.

One analogy I have used successfully is to the "Suggestion mode" of GoogleDocs, describing the PR workflow as similar to suggesting edits to somebody else's document: it is more polite, easier to keep track of what changes are being made and by whom, and provides a quality control step.

I also think this is a part of the curriculum that could really benefit from an illustration to share during training.

### Which part of the content does your suggestion apply to?

https://carpentries.github.io/lesson-development-training/collaborating-team.html#managing-contributions

### PR DESCRIPTION
**Pull request by anenadic**
Fixes #463. 
Will crate a separate issue to create a diagram for a PR collaboration workflow.

### CODE REVIEW

File: episodes/collaborating-team.md [hunk 1]
```diff
@@ -104,6 +104,16 @@ On GitHub, these units of discussion, review,
...
 
+GitHub pull requests is analogous to tracking changes in a shared online document - 
+both involve proposing, reviewing, and finalising updates collaboratively. 
+In a shared document, edits are submitted for review by a proposer, akin to creating 
+a pull request in GitHub, where changes are proposed via a branch. 
+Reviewers in both cases look at the proposed changes, provide feedback via comments, 
+request modifications, and resolve conflicts before final approval. 
+Once approved, changes are integrated into the main document or `main` branch, 
+with version histories in both systems providing an audit trail of contributions. 
+This structured workflow ensures quality and accountability in collaborative efforts.
```
Comment (tobyhodges) [inline episodes/collaborating-team.md:115]:
```suggestion
Pull requests are analogous to tracking changes in a shared online document - 
both involve proposing, reviewing, discussing, and finalising updates collaboratively. 
In a shared document, edits are submitted for review by a proposer, akin to creating 
a pull request in GitHub, where changes are proposed via a branch. 
Reviewers in both cases look at the proposed changes, provide feedback via comments, 
request modifications, and may need to resolve conflicts between multiple competing versions 
before final approval. 
Once approved, changes are integrated into the main document (the `main` branch in your lesson repsository), 
with version histories in both systems providing an audit trail of contributions. 
This structured workflow ensures quality and accountability in collaborative efforts.
```

**Review by tobyhodges — CHANGES_REQUESTED** [review]
Thanks @anenadic, I like this a lot.

Please could you adjust to use sentence line breaks before we merge?

File: episodes/collaborating-team.md [hunk 2]
```diff
@@ -104,6 +104,17 @@ On GitHub, these units of discussion, review,
 and acceptance/rejection of the changes within a branch
 are called _Pull Requests_.
 
+Pull requests are analogous to tracking changes in a shared online document - 
+both involve proposing, reviewing, discussing, and finalising updates collaboratively. 
+In a shared document, edits are submitted for review by a proposer, akin to creating 
+a pull request in GitHub, where changes are proposed via a branch. 
+Reviewers in both cases look at the proposed changes, provide feedback via comments, 
+request modifications, and may need to resolve conflicts between multiple competing versions
```
Comment (anenadic) [inline episodes/collaborating-team.md:112]:
```suggestion
(optionally) request modifications, and may need to resolve conflicts between multiple competing versions 
```

File: episodes/collaborating-team.md [hunk 3]
```diff
@@ -104,6 +104,17 @@ On GitHub, these units of discussion, review,
...
+both involve proposing, reviewing, discussing, and finalising updates collaboratively. 
+In a shared document, edits are submitted for review by a proposer, akin to creating 
+a pull request in GitHub, where changes are proposed via a branch. 
+Reviewers in both cases look at the proposed changes, provide feedback via comments, 
+(optionally) request modifications, and may need to resolve conflicts between multiple competing versions 
+before final approval. 
+Once approved, changes are integrated into the main document (the `main` branch in your lesson repsository), 
+with version histories in both systems providing an audit trail of contributions. 
+This structured workflow ensures quality and accountability in collaborative efforts.
+
```
Comment (anenadic) [inline episodes/collaborating-team.md:117]: Pull requests are analogous to tracking changes in a shared online document - both involve proposing, reviewing, discussing, and finalising updates collaboratively. 
In a shared document, edits are submitted for review by a proposer, akin to creating a pull request in GitHub, where changes are proposed via a branch. 
Reviewers in both cases look at the proposed changes, provide feedback via comments, (optionally) request modifications, and may need to resolve conflicts between multiple competing versions before final approval. 
Once approved, changes are integrated into the main document (the `main` branch in your lesson repsository), with version histories in both systems providing an audit trail of contributions. 
This structured workflow ensures quality and accountability in collaborative efforts.

**Review by tobyhodges — APPROVED** [review]
Great, thank you!

**Pull request merged**